torch/csrc/distributed/c10d/ProcessGroup.cpp

```diff
@@ -111,74 +107,19 @@ c10::intrusive_ptr<Backend> ProcessGroup::getBackend(
   }
 
   // Get the backend type associated with the device
-  TORCH_CHECK(
-      deviceTypeToBackendType_.find(deviceType) !=
-          deviceTypeToBackendType_.end(),
-      "No backend type associated with device type ",
-      deviceType);
-  ProcessGroup::BackendType backendType =
-      deviceTypeToBackendType_.at(deviceType);
+  ProcessGroup::BackendType backendType;
+  try {
+    backendType = deviceTypeToBackendType_.at(deviceType);
+  } catch (const std::out_of_range& e) {
+    TORCH_CHECK(
+        false, "No backend type associated with device type ", deviceType);
+  }
 
   // Check if the backend has already been initialized
   if (backendTypeToBackend_.find(backendType) != backendTypeToBackend_.end()) {
     auto backend = backendTypeToBackend_.at(backendType);
     deviceTypeToBackend_[deviceType] = backend;
     return backend;
```

Comment:
nit: it seems this unnecessary 2nd search is still here. Can you please remove it later? Thanks!